### PR TITLE
Pass trivially copy types as copy

### PR DESCRIPTION
**Pull request by kjeremy**

### CODE REVIEW

**Comment by lnicola** [comment]
If that's a `clippy` lint, should we enable the rule in `xtask line`?

**Comment by kjeremy** [comment]
There was at least one that I could not apply due to `salsa::cycle`

File: crates/ra_proc_macro_srv/src/proc_macro/mod.rs [hunk 1]
```diff
@@ -201,54 +201,54 @@ impl Span {
     }
 
     /// The original source file into which this span points.
-    pub fn source_file(&self) -> SourceFile {
+    pub fn source_file(self) -> SourceFile {
```
Comment (edwin0cheng) [inline crates/ra_proc_macro_srv/src/proc_macro/mod.rs:204]: This part of code related to libproc-macro abi, maybe leave it as is ?

Comment (kjeremy) [reply crates/ra_proc_macro_srv/src/proc_macro/mod.rs:204]: I can drop changes in `ra_proc_macro_srv` if that's what you prefer.

Comment (edwin0cheng) [reply crates/ra_proc_macro_srv/src/proc_macro/mod.rs:204]: Yeah.

Basically all files inside `ra_rpoc_macro_srv/src/proc_macro/` are ABI related.

**Pull request merged**